src/core/teleop_session_manager/python/teleop_session.py

```
@@ -229,11 +248,65 @@ def step(
+                "teleop_control_pipeline must output 'reset_event' (single bool pulse)"
+            )
+
+        state_group = control_outputs["teleop_state"]
+        reset_group = control_outputs["reset_event"]
+
+        states = list(ExecutionState)
+        if len(state_group) != len(states):
+            raise ValueError(
+                "teleop_control_pipeline output 'teleop_state' must have one bool slot "
+                f"per ExecutionState member ({len(states)} slots expected)"
+            )
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)